extensions/blocks/map/edit.js

```diff
@@ -188,21 +193,23 @@ class MapEdit extends Component {
 							/>
 						</PanelBody>
 					) : null }
-					<PanelBody title={ __( 'Mapbox Access Token', 'jetpack' ) } initialOpen={ false }>
-						<TextControl
-							label={ __( 'Mapbox Access Token', 'jetpack' ) }
-							value={ apiKeyControl }
-							onChange={ value => this.setState( { apiKeyControl: value } ) }
-						/>
-						<ButtonGroup>
-							<Button type="button" onClick={ this.updateAPIKey } isDefault>
-								{ __( 'Update Token', 'jetpack' ) }
-							</Button>
-							<Button type="button" onClick={ this.removeAPIKey } isDefault>
-								{ __( 'Remove Token', 'jetpack' ) }
-							</Button>
-						</ButtonGroup>
-					</PanelBody>
+					{ 'wpcom' !== apiKeySource && (
+						<PanelBody title={ __( 'Mapbox Access Token', 'jetpack' ) } initialOpen={ false }>
+							<TextControl
+								label={ __( 'Mapbox Access Token', 'jetpack' ) }
+								value={ apiKeyControl }
+								onChange={ value => this.setState( { apiKeyControl: value } ) }
+							/>
+							<ButtonGroup>
+								<Button type="button" onClick={ this.updateAPIKey } isDefault>
+									{ __( 'Update Token', 'jetpack' ) }
+								</Button>
+								<Button type="button" onClick={ this.removeAPIKey } isDefault>
+									{ __( 'Remove Token', 'jetpack' ) }
+								</Button>
+							</ButtonGroup>
+						</PanelBody>
+					) }
```

Comment:
How would a user enter their own API key, if we needed them to?

Reply:
We would just have to blacklist the site in the `/mapbox` endpoint.
Then, its permission check would fail and return 403.

With a non-200 code, Service API Keys returns `[ 'key' => '', 'source' => 'site' ]`.
With `source !== 'wpcom'`, the block needs an explicitly provided access token, and shows the whole UI to insert it.

Reply:
A big problem with this approach is that it requires breaking the map functionality (by blacklisting) in order for a user to enter their own key. Considering that higher traffic users are the one's we'd likely ask to enter their own key, it's not a very good experience for them.

Tracking in https://github.com/Automattic/jetpack/issues/14473